src/OSPSuite.Core/Domain/Services/ObservedDataTask.cs

```diff
@@ -166,11 +178,73 @@ public void AddImporterConfigurationToProject(ImporterConfiguration configuratio
          _executionContext.AddToHistory(new AddImporterConfigurationToProjectCommand(configuration).Run(_executionContext));
       }
 
+      public void RemoveUsedObservedDataFromSimulation(IReadOnlyList<UsedObservedData> usedObservedDataList)
+      {
+         if (!usedObservedDataList.Any())
+            return;
+
+         foreach (var usedObservedData in usedObservedDataList)
```

Comment:
this loop is weird. Feels like this is not a loop at all because you return after the first time the dialog is shown
It feels more like a query to see of this is used and then a message ?
My first instinct was to say: BEWARE, showing a message in a loop. But there is the return on line 195